src/creature.cpp

```diff
@@ -207,8 +207,10 @@ bool Creature::sees( const Creature &critter ) const
     if( wanted_range <= 1 &&
         ( posz() == critter.posz() || g->m.valid_move( pos(), critter.pos(), false, true ) ) ) {
         return visible( p );
+    } else if( ( wanted_range > ( ( p->get_per() - ( p->encumb( bp_eyes ) / 5.0f ) ) / 3.0f ) ) && 
```

Comment:
`p` can be null. You must check for this before actually using it. And `p` is the creature that is looked at (the functions checks whether `p` / `critter` can be seen). The visibility of someone should not depend on their perception. (Note: `this` is the creature that is actually looking.)

Reply:
Yeah, I don't really know how to program lol.

Thanks for the help though, it should be working now :)